pkg/datastore/dynamodb/dynamodb.go

```diff
@@ -0,0 +1,122 @@
+package dynamodb
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsdynamodb "github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/grid-x/aws-auto-snapshot/pkg/datastore"
+)
+
+const (
+	primaryKey = "snapshot_resource"
+	rangeKey   = "created_at"
+)
+
+// DynamoDB represents a datastore that uses dynamodb under the hood
+type DynamoDB struct {
+	table  string
+	client *awsdynamodb.DynamoDB
+
+	logger log.FieldLogger
+}
+
+type item struct {
+	Resource  string            `dynamodbav:"snapshot_resource"`
+	CreatedAt int64             `dynamodbav:"created_at"`
+	ID        string            `dynamodbav:"snap_id"`
+	Labels    map[string]string `dynamodbav:"labels"`
+}
+
+// New creates a new DynamoDB-based datastore
+func New(client *awsdynamodb.DynamoDB, table string) *DynamoDB {
+	return &DynamoDB{
+		table:  table,
+		client: client,
+		logger: log.New().WithFields(log.Fields{
+			"component": "datastore",
+			"datastore": "dynamodb",
+		}),
+	}
+}
+
+// StoreSnapshotInfo stores the given snapshot info in the datastore
+func (d *DynamoDB) StoreSnapshotInfo(info *datastore.SnapshotInfo) error {
+
```

Comment:
Please check `info == nil` otherwise it might panics

Reply:
good point